### PR TITLE
Jetpack Lifecycle 2.8.0 -> 2.8.4

### PR DESCRIPTION
**Pull request by igordmn**

### CODE REVIEW

File: lifecycle/lifecycle-common/build.gradle [hunk 1]
```diff
@@ -71,7 +71,7 @@ kotlin {
             dependencies {
                 api(libs.kotlinStdlib)
                 api(libs.kotlinCoroutinesCore)
-                api(project(":annotation:annotation"))
```
Comment (MatkovIvan) [inline lifecycle/lifecycle-common/build.gradle]: Does annotation support all our platforms? It seems as invalid change

File: lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt [hunk 2]
```diff
@@ -118,9 +118,10 @@ fun <T : Any> SavedStateHandle.saveable(
     saver: Saver<T, out Any> = autoSaver(),
     init: () -> T,
 ): PropertyDelegateProvider<Any?, ReadOnlyProperty<Any?, T>> =
-    PropertyDelegateProvider { _, property ->
```
Comment (MatkovIvan) [inline lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt:121]: `qualifiedName` reflection API is not supported yet on web. Please copy `internal expect val <T : Any> KClass<T>.canonicalName` from viewmodel module

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: lifecycle/lifecycle-common/build.gradle (hunk 1 above)
Comment (igordmn) [reply lifecycle/lifecycle-common/build.gradle]: Fixed

File: lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt (hunk 2 above)
Comment (igordmn) [reply lifecycle/lifecycle-viewmodel-compose/src/commonMain/kotlin/androidx/lifecycle/viewmodel/compose/SavedStateHandleSaver.kt:121]: Fixed

**Review by MatkovIvan — APPROVED** [review]
Please update toml (to 2.8.1) too

**Pull request merged**